src/Executables/Auth/PacketHandlers/LoginRequestHandler.cs

```diff
@@ -93,4 +131,35 @@ public async Task ExecuteAsync(AuthPacketContext<LoginRequest> ctx, Cancellation
             Result = 1
         });
     }
+
+    private async Task<bool> CheckExistingConnectionOfAsync(AccountData account)
+    {
+        var isLoggedIn = await _cacheManager.Shared.Exists($"account:token:{account.Id}");
+
+        return isLoggedIn == 1;
+    }
+    
+    private async Task<string?> DecideAlreadyLoggedInStatusAsync(AccountData account)
+    {
+        var attemptKey = $"account:attempt:success:{account.Id}";
+        var accountKey = $"account:token:{account.Id}";
+        // increment the attempts in Shared cache
+        var attempts = await _cacheManager.Shared.Incr(attemptKey);
+        // set expiration on the key
+        await _cacheManager.Shared.Expire(attemptKey, ExpiresIn.OneMinute);
+        
+        // check if the attempts are greater than the limit
+        if (attempts > DropConnectionAfter)
+        {
+            // publish a message through redis to drop the connection
+            await _cacheManager.Publish("account:drop-connection", account.Id);
+            // delete the account key
+            await _cacheManager.Shared.Del(accountKey);
+            await _cacheManager.Shared.Del(attemptKey);
+            
+            return "";
```

Comment:
why not null if the return type is nullable?

Reply:
by definition of `enum.AsString` the function could return null. 

```CSharp
public static string? AsString<TEnum>(this TEnum value, EnumFormat format)
        where TEnum : struct, Enum => Cache<TEnum>.Instance.AsString(ref UnsafeUtility.As<TEnum, byte>(ref value), format);
```

At lines 99-100, empty string represents success. So in order to implement original behavior we need to allow the user after certain amount of successful attempts to login by returning empty string.

Reply:
I see. But then I'd change the return type if never null is returned. The type definition of AsString() can be ignored with `! ` if you set up the enum correctly